# Integration Roadmap for Multiscale Simulation of Spike Protein-Induced BBB Disruption and Neuroimmune Dysregulation

## Introduction  
Long COVID is characterized by persistent pathophysiological changes, including blood–brain barrier (BBB) disruption and neuroimmune dysregulation, that contribute to neurological symptoms (e.g. “brain fog,” autonomic dysfunction) ([Blood–brain barrier disruption and sustained systemic inflammation in individuals with long COVID-associated cognitive impairment | Nature Neuroscience](https://www.nature.com/articles/s41593-024-01576-9#:~:text=Vascular%20disruption%20has%20been%20implicated,Accordingly%2C%20peripheral%20blood%20mononuclear%20cells)) ([
            Autonomic Dysfunction Related to Postacute SARS-CoV-2 Syndrome - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10110930/#:~:text=The%20mechanism%20of%20postacute%20COVID,from%20herpes%20simplex%20and%20infectious)). SARS-CoV-2 spike protein interactions with endothelial cells can trigger molecular cascades – such as RhoA-mediated cytoskeletal changes and matrix metalloproteinase-9 (MMP-9) release – that compromise BBB integrity ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=The%20virus%20promotes%20an%20increase,the%20brain%2C%20potentially%20leading%20to)) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=On%20the%20other%20hand%2C%20DeOre,the%20presence%20of%20protein%20S)). The ensuing endothelial barrier breach allows immune cell infiltration and cytokine entry into the brain, inducing glial activation and inflammatory cytokine cascades ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=action%20of%20furin%20protease,astrocytes%20release%20mediators%2C%20such%20as)) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=TNF%2C%20prostaglandins%2C%20glutamate%2C%20ROS%2C%20and,BBB%20is%20exposed%20to%20the)). This neuroinflammatory environment can disrupt neurovascular coupling (NVC) and autonomic regulation, contributing to cognitive impairment and dysautonomia in long COVID ([
            Neurovascular coupling impairment as a mechanism for cognitive deficits in COVID-19 - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10943572/#:~:text=Neurovascular%20coupling%20,i.e.%20T2D%2C16%20peripheral%20artery)) ([
            Autonomic Dysfunction Related to Postacute SARS-CoV-2 Syndrome - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10110930/#:~:text=The%20mechanism%20of%20postacute%20COVID,from%20herpes%20simplex%20and%20infectious)). Modeling these complex, multiscale processes requires an **integrated computational approach**. We propose a comprehensive roadmap that leverages open-source modeling frameworks at multiple scales – from molecular signaling to cellular and tissue dynamics to neural circuit function – to simulate spike protein-induced BBB breakdown and neuroimmune interactions relevant to long COVID. The following roadmap outlines framework selection, module development, validation, interoperability, deployment, and collaborative integration steps to build a multiscale **“virtual neurovascular unit”** model for long COVID research.

## Framework Selection and Multiscale Capability Mapping  
To capture the system-level dynamics, we map each pathogenic layer to an appropriate open-source simulation platform. The rationale for each framework and its role in the integrated model are:  

- **PhysiCell (Cellular Agent-Based Model):** PhysiCell is an open-source agent-based modeling framework for 2D/3D multicellular systems ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=PhysiCell%20is%20an%20open%20source,of%20any%20type%29%20is)). It efficiently simulates individual cells (e.g. endothelial cells, astrocytes, peripheral immune cells) as agents with rules for proliferation, death, and phenotype changes, and it includes diffusion solvers for extracellular factors (cytokines, viral proteins, etc.). We use PhysiCell to model **endothelial and immune cell dynamics** and **diffusive cytokine fields** in the BBB microenvironment. For example, endothelial cells can be agents that respond to spike protein in the circulation by altering adhesion properties, and leukocytes can be agents chemotaxing along cytokine gradients. PhysiCell’s built-in diffusion-reaction capabilities will handle chemical gradients (oxygen, cytokines) and virus/spike protein distribution in tissue ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=PhysiCell%20is%20an%20open%20source,of%20any%20type%29%20is)). Its performance in large-scale cell simulations (with C++ core and parallelization) suits simulation of thousands of cells interacting in and around a brain microvessel.

- **CompuCell3D (Tissue-Scale Morphodynamics):** CompuCell3D (CC3D) is an open-source multiscale modeling environment that uses the cellular Potts model to simulate cell morphology, adhesion, and movement in tissues ([CompuCell3D - Wikipedia](https://en.wikipedia.org/wiki/CompuCell3D#:~:text=CompuCell3D,Glazier%2C%20Fran%C3%A7ois)) ([CompuCell3D - Wikipedia](https://en.wikipedia.org/wiki/CompuCell3D#:~:text=and%20volume%20and%20surface%20area,and%20motility%20and%20other%20basic)). CC3D supports detailed cell shape changes, cell–cell junction modeling, and reaction-diffusion of chemical fields ([CompuCell3D - Wikipedia](https://en.wikipedia.org/wiki/CompuCell3D#:~:text=Hogeweg%20model%20,enables%20modeling%20of%20cellular%20reactions)) ([CompuCell3D - Wikipedia](https://en.wikipedia.org/wiki/CompuCell3D#:~:text=and%20volume%20and%20surface%20area,and%20motility%20and%20other%20basic)). We incorporate CompuCell3D to explicitly model the **BBB structural integrity**: e.g. simulating a layer of tightly coupled endothelial cells on a basement membrane. Using CC3D’s lattice-based approach, we can represent tight junctions and their disassembly when RhoA signaling is activated by spike protein ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=The%20virus%20promotes%20an%20increase,the%20brain%2C%20potentially%20leading%20to)) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=On%20the%20other%20hand%2C%20DeOre,the%20presence%20of%20protein%20S)). CompuCell3D’s strength in simulating cell adhesion and extracellular matrix (ECM) allows representation of **basement membrane** and its degradation by enzymes like MMP-9. In the integrated model, CC3D will capture the **cell morphological changes** (endothelial retraction, gap formation) and **ECM breakdown** that lead to barrier leakage, complementing PhysiCell’s cell-agent biochemistry with physical cell rearrangements.

- **NetPyNE/NEURON (Neuro-Glial Circuit Modeling):** NetPyNE is a high-level Python interface to the NEURON simulator that facilitates building and parallel simulation of biophysical neural networks ([
            NetPyNE, a tool for data-driven multiscale modeling of brain circuits - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC6534378/#:~:text=Biophysical%20modeling%20of%20neuronal%20networks,in%20functions%20for%20visualization%20and)). It supports data-driven specification of neurons and synapses, and importantly can import/export standardized NeuroML models ([
            NetPyNE, a tool for data-driven multiscale modeling of brain circuits - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC6534378/#:~:text=through%20automated%20batch%20runs%2C%20and,used%20to%20teach%20computational%20neuroscience)). We will use NetPyNE/NEURON to simulate **neural and glial network dynamics** within affected brain regions. For example, a model of a cortical microcircuit or brainstem autonomic circuit can be constructed to study how inflammation alters neuronal firing patterns and neurovascular coupling. Neurons and astrocytes can be represented with biophysical detail (e.g. integrate-and-fire or multi-compartment Hodgkin-Huxley models), and their activity can be modulated by diffusible factors (like elevated extracellular glutamate or cytokines). NetPyNE’s parallel simulation capability enables integration of large neural circuits with the slower tissue-scale model ([
            NetPyNE, a tool for data-driven multiscale modeling of brain circuits - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC6534378/#:~:text=Biophysical%20modeling%20of%20neuronal%20networks,optimize%20and%20explore%20network%20parameters)). This framework addresses the **neurofunctional layer** – how BBB breach and cytokines translate into neural firing disruptions, impaired NVC, or autonomic dysregulation (e.g. altered baroreflex signaling leading to postural tachycardia).

- **OpenCOR/COPASI (Biochemical Signaling Networks):** OpenCOR is an open-source, plugin-based modeling environment for creating and simulating CellML-encoded biological models ([
            OpenCOR: a modular and interoperable approach to computational biology - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC4319394/#:~:text=encoding%20of%20mathematical%20models,general%20philosophy%20behind%20OpenCOR%2C%20as)), while COPASI is an open-source simulator for biochemical networks supporting SBML, ODE and stochastic simulations, and parameter estimation ([COPASI](https://copasi.org/#:~:text=COPASI%20is%20a%20software%20application,biochemical%20networks%20and%20their%20dynamics)). These tools will be used to develop **molecular pathway models** relevant to spike-induced signaling. For instance, we can encode the **spike–ACE2–RhoA signaling cascade** within an endothelial cell, or the NF-κB cytokine production pathway in an astrocyte, as a system of ODEs. COPASI allows us to calibrate these pathways against experimental kinetics (e.g. fitting an MMP-9 expression time-course) and explore steady states. We will integrate such signaling models into the cellular agents via SBML (supported by both PhysiCell and CC3D) ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=written%20in%20systems%20biology%20markup,Similar%20approaches)) ([SBML Solver — CC3D Reference Manual 4.6.0 documentation](https://compucell3dreferencemanual.readthedocs.io/en/latest/sbml_solver.html#:~:text=Being%20a%20multi,menu%20for%20options%20available)). Thus, OpenCOR/COPASI provide the **molecular detail layer** for intracellular networks (e.g. RhoA activation, calcium signaling, oxidative stress pathways ([Omics-based analysis of mitochondrial dysfunction and BBB integrity in post-COVID-19 sequelae | Scientific Reports](https://www.nature.com/articles/s41598-024-82180-6#:~:text=CoV,article%20showcases%20the%20evidence%20that)) ([Omics-based analysis of mitochondrial dysfunction and BBB integrity in post-COVID-19 sequelae | Scientific Reports](https://www.nature.com/articles/s41598-024-82180-6#:~:text=mitochondrial%20dysfunction%20is%20a%20major,inflammation%20and%20BBB%20dysfunctioning%20conditions))) that drive cell behavior in the higher-level simulations.

- **AnyLogic (System-Level Process Integration):** At the system level, we may employ a multi-method simulation approach inspired by AnyLogic. AnyLogic allows combining agent-based, discrete event, and system dynamics modeling in one platform ([Agent-Based Simulation Modeling – AnyLogic Simulation Software](https://www.anylogic.com/use-of-simulation/agent-based-modeling/#:~:text=AnyLogic%20combines%20professional%20discrete,are%20always%20a%20better%20choice)). Although AnyLogic itself is a commercial tool, the concept can be implemented in open frameworks (e.g. via Python with SimPy or Mesa for processes, or using statecharts within the ABM). We will use this approach to **orchestrate the multi-scale model**: for example, managing different simulation timelines (viral exposure event, immune response onset, acute vs. chronic phases) and integrating multi-organ interactions (autonomic feedback between brain and heart). A system-level layer can incorporate **discrete events** (such as an initial infection trigger or therapeutic intervention), and **aggregate variables** (such as blood pressure or systemic cytokine levels) that evolve over longer timescales. By designing a supervisory simulation (using Python or a state machine library) that steps through physiological phases and coordinates data exchange between PhysiCell/CC3D and NEURON, we achieve AnyLogic-like **multi-method integration** ([Agent-Based Simulation Modeling – AnyLogic Simulation Software](https://www.anylogic.com/use-of-simulation/agent-based-modeling/#:~:text=AnyLogic%20combines%20professional%20discrete,are%20always%20a%20better%20choice)). This ensures the **whole-body context** (e.g. persistent systemic inflammation maintained by viral remnants ([Omics-based analysis of mitochondrial dysfunction and BBB integrity in post-COVID-19 sequelae | Scientific Reports](https://www.nature.com/articles/s41598-024-82180-6#:~:text=CoV,article%20showcases%20the%20evidence%20that)) ([Omics-based analysis of mitochondrial dysfunction and BBB integrity in post-COVID-19 sequelae | Scientific Reports](https://www.nature.com/articles/s41598-024-82180-6#:~:text=mitochondrial%20dysfunction%20is%20a%20major,inflammation%20and%20BBB%20dysfunctioning%20conditions))) is represented alongside detailed local tissue modeling.

**Mapping Summary:** Each framework addresses a specific scale: COPASI/OpenCOR for intracellular signaling, PhysiCell/CompuCell3D for cellular and tissue-scale dynamics, NetPyNE/NEURON for neural circuit function, and a system-level orchestrator for higher-order processes. Using specialized open tools at each layer ensures we leverage the unique capabilities needed (e.g. diffusion in tissue vs. spiking neurons) while maintaining interoperability. In the following sections, we outline how to extend these frameworks with custom modules for COVID-specific biology, and how to integrate them into a cohesive simulation of BBB disruption and neuroimmune dysregulation.

## Custom Module Development and Integration Strategies  
Implementing long COVID pathophysiology requires developing specific modules and extensions within the selected frameworks. Key plugin/module development areas include:

- **Spike Protein – ACE2/RhoA Signaling Module:** We will create a detailed model of the spike protein’s interaction with endothelial ACE2 receptors and the downstream RhoA pathway that destabilizes tight junctions ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=The%20virus%20promotes%20an%20increase,the%20brain%2C%20potentially%20leading%20to)) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=On%20the%20other%20hand%2C%20DeOre,the%20presence%20of%20protein%20S)). This can be encoded as an **intracellular SBML sub-model** attached to each endothelial cell agent. For example, upon binding of spike to ACE2 (modeled as a surface receptor event), the SBML module triggers a GTPase cycle activating RhoA. Active RhoA then causes changes in cell phenotype within the ABM: in PhysiCell, it could reduce cell-cell adhesion strength or increase cell contractility (simulating tight junction disassembly) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=On%20the%20other%20hand%2C%20DeOre,the%20presence%20of%20protein%20S)). CompuCell3D can be extended with a **custom Steppable** that reads each cell’s RhoA state and adjusts the cellular Potts model adhesion parameters or adds an external force pulling cells apart. The module will include RhoA negative regulators (e.g. the effect of RhoA inhibitor C3 transferase can be simulated by toggling this module off ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=On%20the%20other%20hand%2C%20DeOre,the%20presence%20of%20protein%20S))). By modularizing spike–ACE2–RhoA signaling, we ensure the model reproduces the experimental finding that spike protein engagement activates RhoA and disrupts junctions ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=The%20virus%20promotes%20an%20increase,the%20brain%2C%20potentially%20leading%20to)) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=On%20the%20other%20hand%2C%20DeOre,the%20presence%20of%20protein%20S)).

- **MMP-9 Secretion and ECM Degradation Plugin:** Spike-protein-exposed endothelial cells and infiltrating immune cells (e.g. neutrophils) can upregulate matrix metalloproteinase-9 ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=The%20virus%20promotes%20an%20increase,the%20brain%2C%20potentially%20leading%20to)) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=elucidate%20the%20possible%20mechanisms%20of,the%20rupture%20of%20the%20BBB)). We will implement an **MMP-9 module** such that cells secrete MMP-9 into the extracellular space in response to inflammatory signals. In PhysiCell, this means adding MMP-9 as a diffusing chemical field with a decay (representing protease turnover). In CompuCell3D, we can leverage the built-in diffusion solver to represent MMP-9 concentration and a **custom plugin** to degrade the basement membrane matrix. For instance, basement membrane can be represented either implicitly as a chemical field (collagen IV concentration) that MMP-9 enzymatically reduces, or explicitly as an ECM fiber object (possibly using an addon like PhysiMeSS for fibers ([PhysiMeSS - a new physiCell addon for extracellular matrix modelling](https://gigabytejournal.com/articles/136#:~:text=PhysiMeSS%20,with%20cells%20and%20other%20fibres))). The module will calibrate MMP-9 kinetics so that increased MMP-9 leads to collagen IV degradation and BBB rupture, consistent with in vivo data ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=The%20virus%20promotes%20an%20increase,the%20brain%2C%20potentially%20leading%20to)) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=elucidate%20the%20possible%20mechanisms%20of,the%20rupture%20of%20the%20BBB)). This plugin thus captures the **transcellular route** of barrier opening (ECM/basement membrane breakdown) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=play%20a%20central%20role%20in,through%20RhoA%20activation%20and%20TJs)) complementary to RhoA’s paracellular route.

- **Cytokine Cascade and Immune Cell Activation:** To simulate the neuroimmune dysregulation, we will incorporate modules for **cytokine signaling and immune cell behavior**. Pro-inflammatory cytokines such as TNF-α, IL-1β, and IL-6 are key mediators released by activated microglia, astrocytes, and infiltrating immune cells ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=action%20of%20furin%20protease,astrocytes%20release%20mediators%2C%20such%20as)) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=TNF%2C%20prostaglandins%2C%20glutamate%2C%20ROS%2C%20and,BBB%20is%20exposed%20to%20the)). We will represent major cytokines as diffusive fields in the tissue (with sources and sinks defined by cell states). Each relevant cell type (endothelium, astrocyte, microglia, peripheral immune) gets an intracellular network (SBML via COPASI) defining how it responds to cytokines: e.g. TNF-α binding triggers NF-κB activation and further cytokine production, IL-6 can induce a feedback loop leading to coagulopathy ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=of%20primary%20proinflammatory%20cytokines%2C%20such,the%20permeability%20of%20the%20BBB)), etc. Additionally, immune cells like T lymphocytes or monocytes will be modeled as agents that extravasate when the BBB is leaky. A **chemotaxis module** will guide these cells to migrate toward high chemokine/cytokine concentrations (using PhysiCell’s chemotaxis functions or CC3D’s haptotaxis mechanisms ([CompuCell3D - Wikipedia](https://en.wikipedia.org/wiki/CompuCell3D#:~:text=and%20volume%20and%20surface%20area,and%20motility%20and%20other%20basic)) ([CompuCell3D - Wikipedia](https://en.wikipedia.org/wiki/CompuCell3D#:~:text=rate%20equations%20%20or%20stochastic,and%20other%20basic%20biological%20mechanisms))). We will script the sequence of immune activation: for example, virus-infected endothelial cells release IFN-γ and TNF which activate microglia; microglia release IL-1β and IL-6 that activate astrocytes and endothelium, creating a cytokine feedback loop ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=action%20of%20furin%20protease,oxide%2C%20which%20are%20implicated%20in)). This will be implemented with a combination of **rule-based triggers** in the ABM (e.g. if a microglial agent detects high viral protein, it enters an "activated" state and starts secreting IL-1, IL-6, TNF) and ODE models for downstream signaling (to capture time delays and thresholds). This cytokine module will reproduce the known “cytokine storm” cycle that contributes to BBB damage ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=of%20primary%20proinflammatory%20cytokines%2C%20such,the%20permeability%20of%20the%20BBB)), providing a mechanistic link between initial viral stimuli and widespread inflammation.

- **Neuro-Glial Interaction Module:** In the brain parenchyma side, we will simulate **neuro-glial interactions under neuroinflammation**. Astrocytes and neurons form the neurovascular unit together with vessels; under stress, astrocytes release neurotoxic mediators (prostaglandins, glutamate, ROS, NO) ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=of%20primary%20proinflammatory%20cytokines%2C%20such,the%20permeability%20of%20the%20BBB)). We will extend the NEURON/NetPyNE model by including *astrocyte nodes* or proxy mechanisms: for example, an astrocyte can be modeled as a compartment that takes up excess glutamate and releases factors affecting neurons and vessels. A custom NEURON mechanism or NetPyNE extension can simulate how elevated cytokines (from the tissue model) reduce astrocytic uptake of glutamate or increase astrocytic release of ATP/adenosine, affecting synaptic transmission. We will also incorporate **microglia** as agents that, once in the brain, release neuroactive substances and modulate synaptic connectivity. The neural circuit model will be extended with parameters that can be modified by inflammation – e.g. increased neuronal excitability due to IL-1β (which has been shown to modulate synaptic plasticity), or reduced inhibitory synapse strength. By coupling these changes to outputs of the cytokine module, we get a **dynamic neuroimmune coupling**: as cytokine levels rise, simulate neural network disturbances like slowed oscillatory activity or impaired NVC.

- **Neurovascular Coupling and Blood Flow Dynamics:** To directly simulate neurovascular coupling disruption, we will introduce a simplified model of **cerebral blood flow regulation**. This could involve an ODE model of arteriole diameter controlled by signals from neurons and astrocytes. Normally, active neurons signal to dilate local arterioles (via astrocyte calcium waves, NO release, etc.), ensuring adequate blood flow ([
            Neurovascular coupling impairment as a mechanism for cognitive deficits in COVID-19 - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10943572/#:~:text=Neurovascular%20coupling%20,i.e.%20T2D%2C16%20peripheral%20artery)). In the model, we will have a variable representing vessel radius in a region, which increases with local neural activity under healthy conditions. Under inflammatory conditions, we simulate **NVC impairment** by reducing the responsiveness of this vessel to neural signals (e.g. endothelial dysfunction due to inflammation prevents dilation ([
            Neurovascular coupling impairment as a mechanism for cognitive deficits in COVID-19 - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10943572/#:~:text=The%20endothelium%20is%20an%20essential,Impairment%20of%20the%20endothelium%20in))). We might implement this as an OpenCOR module (a small ODE system) for hemodynamics, or directly within NEURON via a coupling where each neuron’s firing contributes to a “metabolic signal” that drives perfusion. If resources allow, a multi-compartment model of an arteriole (possibly using an existing published model encoded in CellML) can be plugged in. This NVC module will allow us to observe how chronic endothelial inflammation and oxidative stress attenuate the normal blood flow response to neural activity ([
            Neurovascular coupling impairment as a mechanism for cognitive deficits in COVID-19 - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10943572/#:~:text=Neurovascular%20coupling%20,i.e.%20T2D%2C16%20peripheral%20artery)), contributing to cognitive deficits.

- **Autonomic Circuit and Baroreflex Model:** To address long COVID autonomic dysfunction (e.g. POTS), we will incorporate a **central autonomic network model**. Using NetPyNE/NEURON, we can model key brainstem nuclei (e.g. nucleus tractus solitarius, vagal dorsal motor nucleus) and sympathetic neurons that regulate heart rate and blood vessel tone. We will integrate a simplified **baroreflex feedback loop**: an increase in blood pressure inhibits sympathetic drive (and vice versa). In the presence of neuroinflammation, baroreflex sensitivity might be reduced. We will simulate this by, for example, altering the firing threshold or synaptic connectivity in the baroreflex arc when inflammatory cytokines are high (based on evidence that cytokine-induced neuropathy or vagal nerve damage can impair reflexes ([
            Autonomic Dysfunction Related to Postacute SARS-CoV-2 Syndrome - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10110930/#:~:text=highly%20inflammatory%20state%20with%20cellular,40%20%2C%2021)) ([
            Autonomic Dysfunction Related to Postacute SARS-CoV-2 Syndrome - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10110930/#:~:text=the%20most%20common%20clinical%20presentation,that%20these%20patients%20can%20present))). Additionally, small-fiber neuropathy can be represented by decreasing the conduction or output of certain autonomic fibers in the model ([
            Autonomic Dysfunction Related to Postacute SARS-CoV-2 Syndrome - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10110930/#:~:text=highly%20inflammatory%20state%20with%20cellular,40%20%2C%2021)). A simple systemic circulation model (possibly using an AnyLogic-like system dynamics component) will close the loop by computing blood pressure from heart rate and peripheral resistance, which then feeds back to the baroreceptor firing rate. This autonomic module will enable the simulation of phenomena like orthostatic intolerance: e.g. when the model “stands up” (an event causing blood pooling), we can see if the heart rate increase is exaggerated (as in POTS) due to impaired autonomic control. 

Each of these modules will be developed as standalone components first, with careful documentation of inputs/outputs, to facilitate later integration. By using open standards (SBML for biochemical networks, NeuroML for neural components) we ensure these modules can plug into the larger framework with minimal custom coding. **In summary,** the development strategy focuses on modular design: each critical process (spike–RhoA signaling, MMP-9 action, cytokine network, etc.) is implemented as a distinct, testable submodel. This modular approach follows best practices from prior multiscale model efforts ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=To%20accomplish%20this%2C%20we%20use,See%20Box%201)) ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=,integrated%20model%20or%20other%20submodels)), enabling parallel development and future swapping/upgrading of submodels without rewriting the whole system. The integration of these modules is discussed next, along with how we will validate the combined model.

## Model Validation and Calibration Strategies  
Building confidence in the multiscale model requires rigorous validation against experimental and clinical data. We will pursue a multi-pronged validation strategy:

1. **Qualitative Validation of Emergent Behaviors:** We will first verify that the integrated model qualitatively reproduces known phenomena of SARS-CoV-2 neuropathology. For example, the model should show that introducing spike protein increases BBB permeability via both paracellular (tight junction) and transcellular (basement membrane) routes ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=play%20a%20central%20role%20in,through%20RhoA%20activation%20and%20TJs)). We will simulate an **Evans Blue dye extravasation experiment** in silico (dye as a diffusing tracer) and confirm that spike exposure leads to tracer leakage akin to what was observed in SARS-CoV-2 infected hamsters ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=elucidate%20the%20possible%20mechanisms%20of,the%20rupture%20of%20the%20BBB)). Another qualitative check: upon spike protein exposure, endothelial cells in the model should upregulate ACE2 and show RhoA activation with junction disassembly, matching cell-culture BBB chip experiments ([SARS-CoV-2 Spike Protein Disrupts Blood-Brain Barrier Integrity via RhoA Activation - PubMed](https://pubmed.ncbi.nlm.nih.gov/34687399/#:~:text=The%20SARS,a%20key%20molecule%20in%20regulating)) ([SARS-CoV-2 Spike Protein Disrupts Blood-Brain Barrier Integrity via RhoA Activation - PubMed](https://pubmed.ncbi.nlm.nih.gov/34687399/#:~:text=endothelial%20cytoskeleton%20and%20tight%20junction,19%20patients)). We will also look for the emergence of a cytokine-positive feedback loop (“storm”) in the model when infection is severe ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=of%20primary%20proinflammatory%20cytokines%2C%20such,the%20permeability%20of%20the%20BBB)), and that microglial and astrocytic activation leads to neuronal dysfunction (e.g. sustained increased firing or network oscillation changes) consistent with neuroinflammation.

2. **Quantitative Calibration with *in vitro* and *in vivo* Data:** Each module will be calibrated using published data. For the RhoA module, we have quantitative data on RhoA activation timing and the protective effect of RhoA inhibition ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=On%20the%20other%20hand%2C%20DeOre,the%20presence%20of%20protein%20S)); we can tune the kinetics so that model inhibition of RhoA yields restoration of junction proteins (ZO-1) consistent with experiments ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=On%20the%20other%20hand%2C%20DeOre,the%20presence%20of%20protein%20S)). The MMP-9 secretion rates will be adjusted to match concentrations found in patient CSF or serum (e.g. long COVID patients have elevated MMP-9 levels ([Alteration of the blood-brain barrier by COVID-19 and its implication ...](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=,9))). If available, time-course data of MMP-9 in the brain after viral exposure (from animal models) will guide setting secretion and decay rates such that the model’s basement membrane degradation occurs in the same timeframe. We will utilize public **transcriptomic and proteomic datasets** from COVID/autopsy brains ([Omics-based analysis of mitochondrial dysfunction and BBB integrity in post-COVID-19 sequelae | Scientific Reports](https://www.nature.com/articles/s41598-024-82180-6#:~:text=CoV,article%20showcases%20the%20evidence%20that)) ([Omics-based analysis of mitochondrial dysfunction and BBB integrity in post-COVID-19 sequelae | Scientific Reports](https://www.nature.com/articles/s41598-024-82180-6#:~:text=mitochondrial%20dysfunction%20is%20a%20major,inflammation%20and%20BBB%20dysfunctioning%20conditions)) to calibrate the cytokine network: for instance, if patient data show a 3-fold increase in IL-6 mRNA in brain endothelium, we adjust cytokine production parameters so that severe infection in the model yields a similar fold-change in IL-6. COPASI’s parameter estimation tools can fit model parameters to these omics-derived target values. On the neural side, if electrophysiological data (animal or clinical) indicate, say, decreased coherence in EEG alpha band in brain fog patients, we check if our inflamed neural network exhibits a similar reduction in synchronous activity and adjust synaptic weights accordingly.

3. **Cross-Validation with Imaging Biomarkers:** We will compare model outcomes with neuroimaging findings from long COVID cohorts. A prime example is **Dynamic Contrast-Enhanced MRI (DCE-MRI)** evidence of BBB leak in long COVID patients ([Blood–brain barrier disruption and sustained systemic inflammation in individuals with long COVID-associated cognitive impairment | Nature Neuroscience](https://www.nature.com/articles/s41593-024-01576-9#:~:text=long%20COVID%2C%20yet%20it%20is,Accordingly%2C%20peripheral%20blood%20mononuclear%20cells)) ([Blood–brain barrier disruption and sustained systemic inflammation in individuals with long COVID-associated cognitive impairment | Nature Neuroscience](https://www.nature.com/articles/s41593-024-01576-9#:~:text=infection%20and%20in%20patients%20with,Together%2C%20our%20data%20suggest%20that)). The model can simulate an MRI contrast agent introduced to the virtual bloodstream and measure its leakage into brain tissue. We will validate that, in conditions corresponding to long COVID (persistent inflammation, months after infection), the simulated BBB leakage is above a threshold analogous to what MRI detected. Additionally, long COVID brain MRI or PET studies that show neuroinflammation (e.g. via TSPO tracers for microglial activation) can be used: the model’s microglial activation level can be aggregated into a comparable metric for validation. **Neurovascular coupling can be validated** against fMRI or fNIRS data: reports suggest impaired NVC in post-COVID cognitive impairment ([
            Neurovascular coupling impairment as a mechanism for cognitive deficits in COVID-19 - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10943572/#:~:text=Neurovascular%20coupling%20,10%20NVC%20is%20responsible)) ([
            Neurovascular coupling impairment as a mechanism for cognitive deficits in COVID-19 - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10943572/#:~:text=match%20at%20L725%20Neurovascular%20coupling,specific%20impairment)). We will drive our neural model with a stimulus and see if blood flow response (from the NVC module) is blunted relative to normal; this can be compared to fMRI studies measuring reduced cerebrovascular reactivity in long COVID patients. Agreement with these imaging biomarkers would strengthen the model’s credibility.

4. **Use of Public Datasets and Consortia Data:** We will leverage data from initiatives like NIH RECOVER. For example, RECOVER’s biomarker studies might provide longitudinal cytokine profiles in long COVID patients. We can input these cytokine levels as external driving functions to test if the model, under those inputs, predicts BBB disruption and cognitive symptoms. Alternatively, if the model is run forward, its predicted peripheral cytokine levels and immune cell profiles can be compared to RECOVER patient data as out-of-sample validation. We also plan to validate against **autonomic function tests** reported clinically: if the model’s autonomic module is correct, introducing similar neuropathic changes should reproduce POTS criteria (e.g. >30 bpm heart rate increase on tilt) ([
            Autonomic Dysfunction Related to Postacute SARS-CoV-2 Syndrome - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10110930/#:~:text=the%20most%20common%20clinical%20presentation,that%20these%20patients%20can%20present)) ([
            Autonomic Dysfunction Related to Postacute SARS-CoV-2 Syndrome - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10110930/#:~:text=POTS%2C%20characterized%20by%20partial%20dysautonomia,11)). Any deviations will prompt refinement of that module.

5. **Sensitivity Analysis and Uncertainty Quantification:** Before full validation, we will perform global sensitivity analysis on model parameters to identify which pathways most strongly influence outcomes (e.g. BBB permeability or neuronal firing rate). This helps focus validation on critical parameters. We will then systematically vary those parameters within plausible ranges and see if the model can still match empirical data – providing an uncertainty bound. Parameters that are unknown (due to lack of data) will be treated with uncertainty, and we might propose experimental measurements if the model suggests they are important.

6. **Iterative Refinement with Expert Feedback:** The validation process will be iterative. We will involve domain experts (e.g. neuroimmunologists, neurologists from long COVID clinics) to qualitatively judge whether model outputs “make sense.” For instance, does the timing of events in simulation mirror clinical progression (acute infection, subacute phase, chronic phase)? If the model predicts an outcome contrary to observations (say, an unrealistically high level of cell death), we will trace which module is responsible and refine its rules or parameters. This continuous calibration loop ensures the integrated model remains grounded in reality at each scale.

For every submodel, we will maintain test cases (e.g. a standalone simulation of just the RhoA signaling in a single cell, or just the neural network without inflammation) and compare those against known results or analytical solutions. Model validation will culminate in an **integrated test**: simulating a scenario of viral infection and checking a suite of outcomes against multiple data sources (histology, imaging, blood biomarkers, clinical symptoms). Only after the model passes these comparisons (within reasonable error) will we consider it validated for hypothesis testing. By leveraging public datasets and the literature, the model will be both **calibrated and validated**, increasing confidence that its predictions about long COVID mechanisms are reliable.

## Interoperability and Integration Strategies  
Connecting the diverse modeling components into a single coherent simulation is a core challenge. We will adopt standardized model description formats and middleware to ensure interoperability:

- **SBML for Biochemical Network Exchange:** We will encode all intracellular signaling pathways (e.g. endothelial RhoA activation, astrocyte NF-κB cascade) in the Systems Biology Markup Language (SBML) format, which is widely supported ([
            OpenCOR: a modular and interoperable approach to computational biology - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC4319394/#:~:text=by%20an%20interested%20party%29,%282015)) ([
            OpenCOR: a modular and interoperable approach to computational biology - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC4319394/#:~:text=experiments%2C%20etc,%282015)). PhysiCell’s and CompuCell3D’s SBML integration capabilities (via libRoadRunner) allow each cell agent to host an SBML model that updates each time step ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=written%20in%20systems%20biology%20markup,Similar%20approaches)) ([SBML Solver — CC3D Reference Manual 4.6.0 documentation](https://compucell3dreferencemanual.readthedocs.io/en/latest/sbml_solver.html#:~:text=Being%20a%20multi,menu%20for%20options%20available)). This means we can design and test a signaling pathway in COPASI or OpenCOR, export it as `model.xml` (SBML), and then attach it to cells in the ABM. The agent-based simulators will call the SBML solver internally to evolve those ODEs alongside their spatial updates. Using SBML as a bridge guarantees that the biochemical dynamics remain consistent across platforms and can be modified without recompiling the whole simulation. For instance, if new data suggests altering the IL-6 signaling network, we edit the SBML file, and the change propagates to all cells in the next run. This decouples molecular model development from the tissue simulation code.

- **NeuroML/SONATA for Neural Network Models:** We will leverage NeuroML (Neural Model Markup Language) to define the neuronal circuit structure and cell properties ([
            NetPyNE, a tool for data-driven multiscale modeling of brain circuits - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC6534378/#:~:text=through%20automated%20batch%20runs%2C%20and,used%20to%20teach%20computational%20neuroscience)). NetPyNE can export the built network to NeuroML and also import networks specified in NeuroML, ensuring our neural model is **shareable and standardized**. This is useful if we want to replace our neural model or have others use it independently. We can also use SONATA (a format for large-scale network model connectivity) via NetPyNE ([
            NetPyNE, a tool for data-driven multiscale modeling of brain circuits - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC6534378/#:~:text=NetPyNE%20tool%20%28www,traces%2C%20spike%20raster%20plots%2C%20local)) if needed. **Integration with the tissue model** will occur through a *co-simulation approach*: the ABM (PhysiCell/CC3D) runs on a slower time scale (minutes), while the neural model (NEURON) runs at milliseconds. We will implement a synchronization scheme where at each coarse time step (e.g. 1 second), the tissue simulator passes relevant data to the neural simulator and vice versa. For example, every second of simulation, the average concentration of key cytokines or neurotransmitters in the brain tissue model will be sent to the neural model (perhaps via a Python API call or shared memory). The neural model will adjust neuron parameters (e.g. increase spontaneous firing rate if cytokine = X). Conversely, neural activity data (like if a certain region is highly active or a seizure-like event occurs) can influence the tissue model – perhaps causing local blood flow changes or stress signaling. We will explore using the **MUSIC framework** (Multisim Integration Toolkit) or custom Python scripts to handle real-time communication between NEURON and the ABM if tight coupling is necessary. However, if time-scale separation is large, a simpler approach is to run the tissue model and neural model sequentially in chunks (e.g. simulate 1 second of tissue, freeze, update neural model for that second, then continue).

- **Common Data Formats and Middleware:** All simulation components will output data in common formats such as HDF5 or CSV for time-series and VTK for spatial fields. We will maintain a **unified time axis** (likely in simulation seconds) and use real units (e.g. nM for concentrations, mmHg for pressure) to ease interpretation and coupling. The orchestrator (AnyLogic-like supervisor) will likely be a Python workflow that calls each simulation component. For example, we might use Jupyter notebooks or a Python script to instantiate a PhysiCell simulation (through XML config or Python bindings), launch a NEURON simulation (via NetPyNE’s API), and manage their interactions. Python’s flexibility allows using libraries like mpi4py for parallel or distributed runs, or SciPy for analyzing outputs on the fly to decide the next step (if using an event-driven approach). 

- **SBML Solver and BionetSolver in CC3D:** For CompuCell3D specifically, we will utilize its BionetSolver which can directly run SBML models linked to cells ([BionetSolver and CompuCell3D for Multiscale Simulation](https://ryanroper.wordpress.com/#:~:text=Simulation%20ryanroper,and%20solved%20as%20such)) ([SBML Solver — CC3D Reference Manual 4.6.0 documentation](https://compucell3dreferencemanual.readthedocs.io/en/latest/sbml_solver.html#:~:text=Being%20a%20multi,menu%20for%20options%20available)). This means any intracellular network we create can run inside CC3D’s cells as well, maintaining consistency with PhysiCell’s approach. We will ensure that both ABM platforms (PhysiCell and CC3D) use the **same SBML definitions** for shared pathways (e.g. if we simulate BBB endothelial cells in CC3D and peripheral immune cells in PhysiCell, and both involve an NF-κB pathway, we use one SBML model to define it uniformly). This interoperability via SBML ensures **model coherence** across domains.

- **Data Exchange via Standard Protocols:** If the co-simulation becomes complex, we can consider higher-level integration frameworks. For instance, the **Functional Mock-up Interface (FMI)** standard could wrap some of our models (OpenCOR can act as an FMI for ODE systems, and there are efforts to wrap NEURON models as FMUs). This would allow a master algorithm to run all components in lockstep. Alternatively, we can use file-based coupling: e.g. the tissue model dumps a file with current cytokine levels, the neural model reads it at set intervals. Given potential performance issues, we might implement direct memory sharing or utilize a common language interface. Because NetPyNE, PhysiCell, and CC3D all have Python interfaces or can be embedded in Python, our plan is to use **Python as the glue** language. We can create a Python class for the “BBB model” (wrapping PhysiCell/CC3D) and one for the “Neural model” (wrapping NetPyNE), then step them in a single Python program. This avoids complicated external coupling and takes advantage of in-process communication.

- **Temporal Coordination:** Different processes operate on different timescales (e.g. molecular signaling seconds to minutes, cellular changes minutes to hours, neural ms to seconds). We will likely adopt a **multiple-timescale integration**: the neural network might be simulated in bursts (micro time steps) whenever needed to update the state for the slower tissue model. For example, if a surge of cytokines occurs over 10 minutes, we might simulate the neural network at baseline and at the peak to see how firing patterns change, rather than continuously simulate every millisecond for the entire 10 minutes (which would be computationally expensive). This quasi-steady-state coupling assumes neural dynamics equilibrate quickly relative to the progression of inflammation – a reasonable assumption for many aspects of acute vs. chronic changes. In scenarios where fast feedback is needed (like baroreflex which is second-to-second), we may run a short NEURON simulation each time step of the systemic model to update heart rate output. These design choices will be tested for stability and accuracy, adjusting the synchronization interval as needed to capture important feedback without unnecessary overhead.

- **Modularity and Replaceability:** Thanks to using standards (SBML, NeuroML) and a modular approach, each part of the model can be swapped out. For instance, if the community develops a better neural model of COVID brain fog, we can import it via NeuroML. Or if a new cell type needs to be added (say pericytes in the BBB model), we can include them in CC3D without affecting the neural model. Our documentation will specify the interface for each module (e.g. what variables it expects/provides). This design is in line with the **open modular modeling** philosophy advocated in community-driven efforts ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=,integrated%20model%20or%20other%20submodels)) ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=,and%20verifiable%2C%20allowing%20parallel%20development)), aiming to maximize reuse and extensibility. 

- **Visualization and Analysis Integration:** We will integrate analysis pipelines so that results from each component can be combined. For example, we might produce a time-series plot that overlays the model’s predicted serum cytokine levels and a patient’s cytokine data for validation, or a spatial map of model BBB permeability alongside MRI data. Using Python, we can gather data from all submodels and use libraries like Pandas for analysis and Matplotlib for visualization, facilitating interpretation of this complex model.

In summary, interoperability is achieved by **speaking common languages** (SBML, NeuroML, standard data files) and by using a central orchestrator (Python) to manage simulations. This approach will allow the multiscale model to function as a single entity despite its heterogeneous parts, and will make it easier for collaborators to run or modify parts of the model without dealing with the full complexity.

## Deployment and Execution Environments  
Deploying the integrated simulation for practical use will require accommodating its computational demands and ensuring accessibility for researchers. We outline deployment options:

- **High-Performance Computing (HPC) Clusters:** The full multiscale simulation (with thousands of agents and spiking neuron networks) will be computationally intensive. We will target HPC environments for large-scale runs. PhysiCell and CompuCell3D can utilize multiple cores (PhysiCell via OpenMP threads, CC3D via multi-threaded steppables and MPI options) to simulate large tissues efficiently. NetPyNE/NEURON natively supports parallelization of network simulations across cores or nodes ([
            NetPyNE, a tool for data-driven multiscale modeling of brain circuits - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC6534378/#:~:text=NetPyNE%20tool%20%28www,in%20functions%20for%20visualization%20and)). We will prepare **MPI-enabled runs** where, for example, different tissue regions or different batches of cells are handled by separate processes, and NEURON can distribute cells across processors. A typical deployment might involve an HPC job script that loads required modules (e.g. NEURON, MPICH) and executes the orchestrator script under `mpirun`. For instance, CPU cores can be split such that some cores handle the neural simulation threads while others handle the ABM threads, synchronized periodically. We will test scaling (strong and weak scaling) to ensure we can simulate, say, a 1 cm³ tissue volume or 100,000 neurons if needed. For HPC portability, we will containerize the software (see below) and use job arrays for parameter sweeps or uncertainty quantification runs. If GPU acceleration becomes beneficial (e.g. for ODE solving or large matrix operations in NEURON), we will explore GPU-enabled versions of these tools (some NEURON versions support GPUs, and custom CUDA implementations could be integrated for diffusion).

- **Cloud Computing:** To make the model accessible to a wider community and to leverage elastic scaling, we will deploy on cloud platforms as well. This could involve using services like Amazon Web Services or Google Cloud to spin up virtual machines or Kubernetes clusters pre-configured with the simulation environment. We will create a **Docker image** (or Singularity container for HPC) containing all necessary frameworks (PhysiCell, CC3D, NEURON, OpenCOR, COPASI, Python libraries). This ensures reproducibility – anyone running the container has the same software versions and dependencies. On the cloud, one could deploy the container on a powerful VM or a cluster and run simulations without local installation hassles. We might also utilize cloud HPC offerings or cloud scheduler systems to run large experiments, which is useful for collaborative projects where different teams can launch runs on a shared cloud account. Cost management will be considered (e.g. using spot instances or scheduling to run only when needed).

- **JupyterHub and Interactive Environments:** For educational and collaboration purposes, we plan to set up a **JupyterHub** or similar multi-user notebook server on either a cloud VM or institutional server. This will allow researchers (who may not be expert programmers) to interact with the model through notebooks. We will develop Jupyter notebooks with intuitive interfaces – for example, a notebook where users can set input conditions (viral load, presence of a drug, etc.), then run a simulation (perhaps a reduced-scale version for speed), and visualize outputs (like BBB leak graphs or neural raster plots). JupyterHub would manage user sessions so multiple people can experiment with the model simultaneously without interfering with each other. This lowers the barrier to entry, enabling immunologists or clinicians to “tinker” with the model by simply tweaking parameters in a browser. The notebooks will use the underlying Python API we develop for orchestration, demonstrating how to run various scenarios or analyses.

- **Web-Based Visualization/Interface:** Building on the Jupyter approach, we will consider providing a **simplified web application** as an interface to the model. As suggested in previous community modeling projects, an interactive web app can engage users and disseminate the model ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=,to%20increase%20public%20educational%20benefit)). Using tools like Dash or Streamlit (Python web app frameworks), we can create a front-end where users move sliders (for viral dose, for example) or toggle checkboxes (to “apply” a hypothetical therapy like a RhoA inhibitor), and then see simulation results (plots, maybe small animations of BBB). This app can run on a server with the model container in the back-end executing the simulation. While real-time full simulation might be too slow for a web app, we might precompute certain scenarios or run a simplified model for demonstration. The goal is to provide a **visual, intuitive platform** for stakeholders to explore model predictions, which is valuable for community engagement and education.

- **Reproducible Workflows and Version Control:** Deployment also involves ensuring the entire modeling pipeline is reproducible. All code will be version-controlled (Git) and released on a platform like GitHub. We will use continuous integration (CI) to run basic tests whenever the code is updated (for example, ensuring the SBML models still integrate correctly and a small test simulation runs without error). For computationally heavy integration tests, we can schedule periodic runs on HPC or cloud and compare results to previous runs (identifying any drift due to changes). The container images will be versioned and published (e.g. on Docker Hub or a similar registry) so anyone can pull a specific release of the model. We will also provide **installation scripts and documentation** for those who wish to install everything natively on their systems. This includes instructions for Windows, macOS, and Linux, though due to the complexity, we expect most will prefer the container or HPC modules route.

- **Scalability and Performance Tuning:** As part of deployment, we will profile the integrated model to find bottlenecks. If the neural simulation is dominating runtime, we might reduce neural detail for large runs (e.g. use point-neuron models instead of multi-compartment). If the diffusion solver in CC3D is slow, we could switch to PhysiCell’s solver for that aspect or vice versa, or reduce grid resolution. We may implement adaptive time stepping: larger time steps during periods of little change, smaller steps during fast dynamics (like acute infection onset). HPC deployment allows for parallel parameter sweeps – for example, exploring different what-if scenarios (different variants of virus, etc.) in parallel jobs. 

- **Data Management:** Running large simulations will produce large amounts of data (3D fields, network spike data, etc.). We will deploy a data management plan: using cloud storage or institutional storage to save raw outputs, and providing analysis scripts to distill them into summary metrics. Relevant results will be curated (for example, snapshots of BBB integrity or peak cytokine levels) and perhaps stored in a database for quick query (if we run many scenarios as in an in-silico experiment).

Overall, the deployment strategy ensures that the model can be executed on powerful computing resources for research-grade simulations, while also providing accessible avenues (notebooks, web apps) for broader use and experimentation. The use of containers and JupyterHub will facilitate **collaborative development**, as everyone will be working in identical environments. Furthermore, these deployments set the stage for eventual cloud-based **services or APIs** where external applications (like other models or decision support tools) could request simulations from our model, integrating it into a wider ecosystem.

## Community Collaboration and Integration with Ongoing Initiatives  
A model of this scope will benefit greatly from community involvement and should contribute to broader research efforts on COVID-19. We plan the following steps to foster collaboration and integrate with existing initiatives:

- **Open-Source Release and Collaborative Platform:** All model code, configurations, and documentation will be released under an open-source license (e.g. MIT or CC-BY for documentation). We will host the project on GitHub (or a similar platform like GitLab) to allow issue tracking, community feedback, and contributions. The repository will be structured for clarity (with separate folders for each submodel, SBML files, example scripts, etc.). We will encourage other researchers to use and contribute to the model, for example by providing a contributor’s guide and labeling “good first issues” for new contributors. This open development approach aligns with the community-driven modeling ethos used in earlier COVID modeling efforts ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=,parallel%20studies%20by%20multiple%20labs)) ([
            Rapid community-driven development of a SARS-CoV-2 tissue simulator - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC7239052.3/#:~:text=,source%20for%20maximum%20public%20benefit)).

- **NIH RECOVER Initiative Integration:** We will actively seek collaboration with the NIH RECOVER Initiative, which is a large multi-institution effort to understand and treat long COVID. This could involve presenting our modeling plans and preliminary results to RECOVER working groups focused on neuropathology. By aligning our model outputs with RECOVER’s data collection (for instance, predicting biomarkers that RECOVER is measuring in patients), we can provide a mechanistic interpretation for their findings. Conversely, RECOVER’s longitudinal clinical data can inform our model scenarios (e.g. common timelines of symptom development can shape simulation timelines). We may propose a formal collaboration where our simulation platform is used to test hypotheses generated by RECOVER (for example, testing if a certain cytokine profile is sufficient to cause cognitive impairment, which could then guide what measurements RECOVER emphasizes). If possible, sharing data (under proper agreements) with our modeling team will greatly enhance model validation. Ultimately, we envision the model being a tool within RECOVER to integrate their multi-omics and clinical observations into a coherent mechanistic picture.

- **COVID Modeling Consortia and Working Groups:** We will engage with consortia such as the IMAG/MSM Multiscale Modeling working group on viral pandemics and any specific COVID-19 modeling collaborations. By participating in these forums (presenting updates in workshops, contributing to collaborative papers), we ensure our work is synergistic with others. For example, if another group is modeling a different aspect (like systemic immune response or viral evolution), we can discuss how to connect the models or share parameters. The COVID-19 HPC Consortium and similar initiatives that emerged during the pandemic might offer resources or expert advice on optimizing the model. We will also follow efforts like the **UT COVID-19 Modeling Consortium** for any epidemiological inputs needed, although our focus is mechanistic rather than epidemiologic. Cross-pollinating with others prevents duplication and opens the door to **model interoperability** with external models (perhaps one day linking our model of the brain to another model of lung infection to simulate whole-body outcomes).

- **Contributing to Model Repositories:** To further integrate with the community, we will contribute simplified versions or components of our model to public repositories. For instance, BioModels Database could host the SBML files of our key signaling networks (so other modelers can reuse the calibrated RhoA or cytokine models). Similarly, we can upload the neural network model (if standalone) to Open Source Brain or ModelDB so neuro modelers can inspect or repurpose it. The CellML models (if any developed via OpenCOR) could be shared via the Physiome Model Repository. By doing this, even those not directly using our entire simulation can benefit from parts of it.

- **Collaboration on Module Development:** We recognize that building every aspect ourselves may be time-consuming, and collaboration can accelerate progress. We will reach out to developers of each framework (PhysiCell, CC3D, NetPyNE, etc.) for guidance and possibly co-development of needed features. For example, if we find we need a new feature in PhysiCell (say, a more complex chemotaxis rule), we can coordinate with the PhysiCell community (via their forums or GitHub) to implement it in a general way that benefits all users, not just our project. This two-way collaboration – we contribute code back, and we gain from their expertise – will strengthen the tools and ensure long-term support. We anticipate possibly contributing a **CompuCell3D plugin for dynamic tight junctions** or an extension to NetPyNE for easier cytokine modulation of neurons. These contributions would be documented and submitted to the respective projects.

- **Community Verification and Benchmarks:** We will encourage other scientists to run our model (or components) and verify results. Perhaps we will organize a **model hackathon or workshop** (virtually) where we walk through the code and have participants run simulations, adjusting parameters to see if they can “break” the model or identify weaknesses. This can reveal edge cases and increase robustness. Such a collaborative exercise also trains new users and possibly recruits them to help extend the model (for example, someone might add a missing cell type or process that they specialize in). We may also compare our model’s predictions with those of other published models (if any) of COVID neuroinflammation. If discrepancies arise, we will investigate alongside the community.

- **Linking with Experimental Initiatives:** Beyond modeling consortia, we plan to integrate with experimental efforts. For example, if there are ongoing **in vitro BBB organoid experiments for COVID** or **animal models of long COVID neuroinflammation**, we can propose that our model be used to help interpret those experiments. In return, those labs might share data for validation. This kind of collaboration grounds the model in experimental reality and ensures it stays biologically plausible. It also creates a feedback loop: model predictions can suggest new measurements for experiments (e.g. “the model predicts RhoA remains active chronically; can we measure RhoA activity in long COVID brain tissue?”).

- **Publication and Dissemination:** We will publish the integrated model and results in open-access journals, which will increase visibility and invite feedback from the scientific community. Accompanying the paper, all model materials will be available (following the good practice of computational model sharing). We will highlight how our model aligns with and contributes to initiatives like NIH RECOVER (e.g. citing that connection) to show that it’s a community resource. Moreover, we will continue to update the model post-publication as new data emerges (possibly maintaining a live “living model” document or version 2, 3, etc.). Community suggestions will be tracked on our GitHub issue list and addressed in updates.

- **Long-Term Maintenance and Collaboration:** We envision this model as part of a sustained collaborative project. As long COVID research evolves (e.g. new variants, new treatments, or new findings about mechanisms), we will integrate those. We will invite collaborators to be co-developers – for example, one group might take on refining the autonomic dysfunction part, while another focuses on adding the effect of specific therapeutics (like corticosteroids or neuroprotective drugs). We will maintain a core team to review contributions for quality and consistency. Ideally, this becomes an **open consortium for multiscale neuro-COVID modeling**, bridging computational and experimental experts. We will seek funding for such collaborative work to ensure continuity (possibly via NIH or NSF programs that support multi-group projects).

By engaging with existing large-scale efforts (RECOVER, consortia) and fostering an open collaborative environment, we aim to make this multiscale model a **community asset**. The roadmap is not just about technical integration, but also about integrating knowledge and people – uniting the community’s expertise to tackle the complex problem of long COVID’s neurological impact. This collaborative spirit increases the model’s credibility, accelerates its development, and ultimately maximizes its scientific and clinical impact.

## Implementation Timeline and Milestones  

To execute this roadmap, we propose a phased timeline spanning approximately 3 years, with clear milestones at each stage:

### Phase 1 (Months 0–6): Requirements Gathering and Framework Setup  
**Tasks:** Form the core development team and consult with domain experts to finalize model scope and requirements. Define the specific hypotheses and use-cases (e.g. simulating brain fog mechanisms, testing a drug intervention). Set up the software environment: install PhysiCell, CompuCell3D, NEURON/NetPyNE, COPASI, OpenCOR, and necessary Python libraries. Establish version control (GitHub repo) and continuous integration for basic tests. **Milestones:** By month 3, complete a detailed design document mapping biological processes to model components (as partly outlined above). By month 6, have each framework running in isolation with simple example models (e.g. a basic PhysiCell simulation of cells and diffusion, a NEURON network of a few neurons) to ensure the tools are functioning. Also, create initial SBML models for a couple of key pathways (perhaps a prototype of the RhoA module). 

### Phase 2 (Months 6–12): Module Development and Unit Testing  
**Tasks:** Begin implementing the custom modules. In parallel sub-teams: (a) develop the **spike-ACE2-RhoA signaling SBML** and test it in COPASI/OpenCOR for correctness (month 6–8); (b) implement an MMP-9 secretion and ECM degradation in a simplified CC3D setup (e.g. a sheet of cells with ECM underneath) to see BBB breach in a toy scenario (month 7–9); (c) build a prototype cytokine network model (perhaps TNF-driven feedback) and test its dynamics against known behavior (month 8–10); (d) construct an initial small neural network in NetPyNE (like a few excitatory/inhibitory neurons) and write a script to modulate it with an external “cytokine” variable (month 9–11). Write unit tests for each module – for example, a test that adds spike protein to an endothelial cell model and checks that RhoA goes high and junction protein goes low. **Milestones:** By month 12, we expect to have all major **submodels functioning independently**. This includes verification of SBML models (by comparing simulation output to analytical solutions or literature data) and visualization of each in isolation (e.g. plotting RhoA and MMP-9 over time for a single cell). We should also have decided on the primary simulation platform for the tissue (PhysiCell vs CC3D or a hybrid) based on experiments with both.

### Phase 3 (Months 12–18): Integration of Multiscale Components (Alpha Version)  
**Tasks:** Set up the initial **integration framework**. Begin with coupling the cellular ABM and the molecular models: integrate SBML models into PhysiCell and/or CC3D (month 12–14). Ensure that an endothelial cell agent in the simulation can carry the RhoA and cytokine-production networks and respond to diffusing spike fields appropriately. Next, integrate the neural module: use Python to run a NetPyNE simulation in tandem with a PhysiCell simulation (month 14–16). Develop the orchestrator script that passes data between the two (e.g. update neural parameters based on current cytokine levels). Implement a basic NVC model (perhaps a simple function linking neural activity to a global flow variable) to connect neural output back to the tissue (month 16–17). During this phase, we will likely iterate on data exchange methods and possibly simplify some aspects to achieve a working coupling. We will also incorporate a **basic autonomic model** (even if placeholder) to keep that component in view. **Milestones:** By month 18, the **alpha version of the integrated model** should run end-to-end for a simple test case (for instance, simulate 1 hour of real time in a small tissue volume with a handful of neurons). Milestone deliverables include a demonstration where spike protein is added and we see BBB opening in the ABM and some effect on neural firing in the coupled network. This version will be crude and possibly not quantitatively accurate, but it proves the pipeline works. We will document this milestone by writing a short internal report and maybe a blog post on our GitHub describing the integration progress.

### Phase 4 (Months 18–24): Calibration and Refinement (Beta Version)  
**Tasks:** With an alpha model in hand, focus on **calibrating parameters and refining dynamics**. Use experimental data to adjust the submodels: calibrate the cytokine release rates, MMP-9 effects, etc., as described in the validation plan. This might involve running many simulations with varied parameters (hence possibly using HPC for parameter sweeps) (month 18–21). Improve model fidelity: add missing cell types or mechanisms discovered to be important (e.g. incorporate pericytes in BBB if needed, or add oxidative stress effect on neurons as per new literature) (month 20–22). Enhance the neural model if it was very simplified; possibly scale it up to more neurons or add region-specific cell types to represent, say, the prefrontal cortex vs. brainstem if needed for different symptoms (month 22–24). Start building output analysis tools to compare model output with data (month 20–24), such as scripts to compute “leakage index” or cognitive score proxies from simulation results. **Milestones:** By the end of month 24, have a **beta-version integrated model** that is much closer to realistic behavior. We expect this version to reasonably match a set of validation targets (e.g. it qualitatively matches the timeline of BBB disruption and cytokine peaks seen in severe COVID). We also aim for this version to handle a larger scale (maybe a 3D section of tissue with dozens of cells and a network of tens of neurons) for more realistic heterogeneity. We will likely submit a conference abstract or preliminary paper around this time with the beta model results to get feedback from peers. Additionally, by month 24, the basic deployment setup should be ready: a container that can run the beta model, tested on a cloud instance.

### Phase 5 (Months 24–30): Validation and Verification (Release Candidate)  
**Tasks:** Conduct thorough **validation studies** as planned. This means running the model under various scenarios and comparing outputs with datasets: simulate acute infection vs. long-term chronic phase, simulate presence vs. absence of certain pathways (like knocking out MMP-9 in silico to see if BBB stays intact, as per experimental MMP-9 inhibitor studies). Gather all results and quantify the model’s accuracy. Any discrepancies found will lead to targeted tweaks. By now, we’ll incorporate more data if available (for example, integrating any new RECOVER findings up to 2025). Also, perform **sensitivity analysis** formally (month 24–26) to identify which parameters cause the most variance in outputs. Use this to inform which aspects need more precise calibration or which might need additional biological realism. Parallel to validation, work on **user interface and accessibility**: develop the Jupyter notebooks and possibly the web dashboard for interacting with the model (month 26–29). During this phase we also focus on optimizing performance: ensure that the model doesn’t take unreasonable time or memory for the target problem size, making code optimizations as needed (month 28–30). **Milestones:** By month 30, aim to have a **release candidate v1.0** of the model. This version is fully validated against known benchmarks and is deemed stable. We will write extensive documentation at this stage: a user manual describing how to run the model, descriptions of each parameter, and example test cases. We will likely make the repository public at this point if it wasn’t already, and perhaps issue a call for external testing (beta testers) from the community. The model environment (container, etc.) should be finalized and tested on at least one HPC system and one cloud provider by this time, smoothing out any deployment kinks.

### Phase 6 (Months 30–36): Dissemination, Collaboration, and Expansion (v1.0 Release)  
**Tasks:** Finalize and **publish the model and results**. Prepare a journal manuscript describing the model, its validation, and key insights about long COVID pathophysiology that emerge from simulations. Submit to a peer-reviewed journal (targeting an open-access journal in computational biology or neuroimmunology) around month 32. Address reviewer comments, aiming for acceptance by month 36. In parallel, engage with the community: host a webinar or workshop (month 31–33) demonstrating the model, possibly in coordination with NIH RECOVER or an IMAG workshop. Provide training sessions if needed for interested research groups. As collaborations form (for example, someone might want to use the model to test a new hypothesis), support those efforts by perhaps adding minor features or helping integrate their data (month 30–36). Start integrating any **new initiatives or data** that have arisen – for instance, if by this time new therapies are in trials for long COVID, simulate their potential impact by modifying model parameters (this could be an extension study). Also, use the model to generate **predictions** and new hypotheses: e.g. predict that blocking IL-6 signaling would restore NVC by X%, or that patients with certain genetic backgrounds might have exaggerated responses. Share these findings with experimental collaborators to spur targeted experiments. **Milestones:** The official **v1.0 release** of the integrated model occurs (likely aligning with the publication). All materials are released, including any datasets or scripts for figures. Achieve a stable, documented codebase that the community can start building upon. At this point, measure success by adoption: we might have, say, a few external groups downloading and running the model, or referencing it in their work. Internally, we will have a roadmap for post-36 month activities (like a grant proposal for further work to extend the model to other diseases or to a personalized medicine tool, etc.).

Throughout all phases, we will keep an agile approach: regularly (e.g. bi-weekly) evaluate progress, adjust tasks as new data or challenges arise, and maintain communication across sub-teams. By following this timeline, we ensure steady progress from conceptualization to a validated, deployable simulation platform. Each phase builds on the last, culminating in a comprehensive multiscale model of spike-induced BBB disruption and neuroimmune dysregulation that is ready for use in understanding and combating long COVID. 

**Timeline Summary:** This structured schedule provides a clear path: initial setup and prototyping (Phase 1–2), integration (Phase 3), refinement and calibration (Phase 4), validation (Phase 5), and deployment and community release (Phase 6). While ambitious, this timeline is feasible with dedicated effort and collaborative support, and it aligns with the urgency of addressing long COVID’s impact. By the end of this roadmap, we aim to deliver a robust simulation tool that researchers and clinicians can use to explore interventions and deepen our mechanistic understanding of long COVID neurobiology, fulfilling the project’s ultimate goal. 

**Sources:** The approach and components described draw upon recent findings of SARS-CoV-2’s effects on the BBB and brain ([
            Alteration of the blood-brain barrier by COVID-19 and its implication in the permeation of drugs into the brain - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC10043238/#:~:text=The%20virus%20promotes%20an%20increase,the%20brain%2C%20potentially%20leading%20to)) ([Blood–brain barrier disruption and sustained systemic inflammation in individuals with long COVID-associated cognitive impairment | Nature Neuroscience](https://www.nature.com/articles/s41593-024-01576-9#:~:text=long%20COVID%2C%20yet%20it%20is,Accordingly%2C%20peripheral%20blood%20mononuclear%20cells)), and leverage the capabilities of open-source modeling platforms ([CompuCell3D - Wikipedia](https://en.wikipedia.org/wiki/CompuCell3D#:~:text=CompuCell3D,Glazier%2C%20Fran%C3%A7ois)) ([
            NetPyNE, a tool for data-driven multiscale modeling of brain circuits - PMC
        ](https://pmc.ncbi.nlm.nih.gov/articles/PMC6534378/#:~:text=Biophysical%20modeling%20of%20neuronal%20networks,in%20functions%20for%20visualization%20and)) to create a multiscale, modular simulation framework. This integration roadmap provides a foundation for uniting disparate biological data into a coherent model, driving forward our understanding of long COVID through computational experimentation.
